std/experimental/allocator/building_blocks/scoped_allocator.d

```diff
@@ -89,6 +89,8 @@ struct ScopedAllocator(ParentAllocator)
         toInsert.prev = null;
         toInsert.next = root;
         toInsert.length = n;
+        assert(!root || !root.prev);
```

Comment:
Quite recently there was a longer discussion on a PR about the ~200 HALTs in Phobos that come without message, but segfault. Maybe we could tell the user at least what has gone wrong?
e.g.  "Invalid allocation. Root node isn't at the top of the stack"